pkg/resolution/resource/crd_resource_test.go

```diff
@@ -0,0 +1,260 @@
+)
+
+// getCRDRequester returns an instance of the CRDRequester that has been seeded with
+// d, where d represents the state of the system (existing resources) needed for the test.
+func getCRDRequester(t *testing.T, d test.Data) (test.Assets, func()) {
+	t.Helper()
+	names.TestingSeed()
+	return initializeCRDRequesterAssets(t, d)
+}
+
+func initializeCRDRequesterAssets(t *testing.T, d test.Data) (test.Assets, func()) {
+	t.Helper()
+	ctx, _ := ttesting.SetupFakeContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	c, informers := test.SeedTestData(t, ctx, d)
+
+	return test.Assets{
+		Logger:    logging.FromContext(ctx),
+		Clients:   c,
+		Informers: informers,
+		Ctx:       ctx,
+	}, cancel
+}
+
+func TestCRDRequesterSubmit(t *testing.T) {
+	testCases := []struct {
+		name                     string
+		inputRequest             *v1beta1.ResolutionRequest
+		expectedRequest          *v1beta1.ResolutionRequest
+		expectedResolvedResource *readOnlyResolutionRequest
+		expectedErr              error
+	}{
+		{
+			name:                     "resolution request does not exist and needs to be created",
+			inputRequest:             nil,
+			expectedRequest:          getResolutionRequestFromFile("testdata/submit.resolution_request.created.yaml"),
+			expectedResolvedResource: nil,
+			expectedErr:              resolutioncommon.ErrRequestInProgress,
+		},
+		{
+			name:                     "resolution request exist and status is unknown",
+			inputRequest:             getResolutionRequestFromFile("testdata/submit.resolution_request.unknown.yaml"),
+			expectedRequest:          nil,
+			expectedResolvedResource: nil,
+			expectedErr:              resolutioncommon.ErrRequestInProgress,
+		},
+		{
+			name:                     "resolution request exist and status is true",
+			inputRequest:             getResolutionRequestFromFile("testdata/submit.resolution_request.true.yaml"),
+			expectedRequest:          nil,
+			expectedResolvedResource: &readOnlyResolutionRequest{getResolutionRequestFromFile("testdata/submit.resolution_request.true.yaml")},
+			expectedErr:              nil,
+		},
+		{
+			name:                     "resolution request exist and status is false",
+			inputRequest:             getResolutionRequestFromFile("testdata/submit.resolution_request.false.yaml"),
+			expectedRequest:          nil,
+			expectedResolvedResource: nil,
+			expectedErr:              resolutioncommon.NewError(resolutioncommon.ReasonResolutionFailed, errors.New("message")),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			d := test.Data{}
+			if tc.inputRequest != nil {
+				d.ResolutionRequests = []*v1beta1.ResolutionRequest{tc.inputRequest}
+			}
+
+			testAssets, cancel := getCRDRequester(t, d)
+			defer cancel()
+			ctx := testAssets.Ctx
+			clients := testAssets.Clients
+
+			resolver := ResolverName("git")
+			request := getRequestFromFile("testdata/request.golden.yaml")
```

Comment:
I'm confused what the difference is between this request and the TC input requests?

Reply:
Sorry, I was being lazy.
The full name `ResolutionRequest` should be used in TC.
I will adjust it immediately.